src/python/pants/backend/core/tasks/prepare_resources.py

```diff
@@ -25,6 +27,12 @@ def register_options(cls, register):
     super(PrepareResources, cls).register_options(register)
     register('--confs', advanced=True, type=Options.list, default=['default'],
              help='Prepare resources for these Ivy confs.')
+    register('--resources-dir-patterns-for-jarring', advanced=True, type=Options.list, default=[],
```

Comment:
Remember that option names are already prefixed with the task name. So the long form of this one would be:
--resources-prepare-resources-dir-patterns-for-jarring